File: 프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int l, int r) {
    string s;
    vector<int> v;
    for(int i = l; i <= r; i++) {
        bool flag = true;
        s = to_string(i);
        for(char c: s) {
            if(c != '0' && c != '5') flag = false;
        }
        
        if(flag) {
            v.push_back(i);
        }
    }
    
    if(v.empty()) v.push_back(-1);
    return v;
}

File: 프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.java
import java.util.*;

class Solution {
    public List solution(int l, int r) {
        List<Integer> list = new ArrayList<>();
        
        for(int i = l; i <= r; i++) {
            String s = i + "";
            boolean flag = true;
            
            for(char c: s.toCharArray()) {
                if(c != '0' && c != '5') flag = false;
            }
            
            if(flag) list.add(i);
        }
        
        if(list.isEmpty()) list.add(-1);
        
        return list;
    }
}